tools/public_api_guard/material/core.md

```diff
@@ -272,6 +272,8 @@ export class MatOption<T = any> implements FocusableOption, AfterViewChecked, On
     id: string;
     get multiple(): boolean | undefined;
     // (undocumented)
+    static ngAcceptInputType_disabled: unknown;
```

Comment:
is this correct? wouldn't we want some type-checking still? e..g `string|boolean|null|undefined` maybe?

Reply:
This is the type extracted from core so we can't really control it. Also since it comes from core, it's not our responsibility how the value will be coerced.

Reply:
It's intentionally set to `unknown` here https://github.com/angular/angular/blob/main/packages/core/src/util/coercion.ts#L21. This was a pre-existing function that was being used in `forms` and `router` which I renamed and exported when I was implementing input transforms: https://github.com/angular/angular/commit/68017d4e75abed78b378dce54f860cfa0d0fa42f#diff-40935ddd388217cdaaf99555de6dd86abfe8add60494e35dad8220ed6f1a1203L10

Reply:
I'm thinking, after this PR, the type checking would accept _anything_, while before this PR, we were:

* limiting `disabled` in templates to `string|boolean|null|undefined`
* and via programmatic assignment (e.g. `.disabled = X`), to `boolean`.

I'd expect the acceptance member to follow the type from `BooleanInput`, like before:
https://github.com/angular/components/blob/main/src/cdk/coercion/boolean-property.ts#L13

Reply:
Yeah I understand, but I don't think it really matters as long as we get a boolean in the end which `booleanAttribute` is guaranteed to do. We can't prevent people from shooting themselves in the foot if they really want to.

Reply:
I'd think we still want to narrow it more, so that clearly wrong cases are properly flagged via type-checking. e.g. `[disabled]="someObject"`.

Reply:
Yeah, not saying we should do it here. I'm just trying to flag this here because it surfaces as a type-checking change here